File: 1-learn-a-language/python/python-basics/task16/create_s3_bucket.py
import boto3

def create_bucket(bucket_name, region=None):
    # Create an S3 client
    s3 = boto3.client('s3', region_name=region)

    # Create the S3 bucket
    if region is None:
        s3.create_bucket(Bucket=bucket_name)
    else:
        s3.create_bucket(Bucket=bucket_name, CreateBucketConfiguration={'LocationConstraint': region})

    print(f"Bucket {bucket_name} created successfully!")

# Example usage:
bucket_name = "my-unique-bucket-name-12345"  # Change this to your desired unique bucket name
region = "us-west-1"  # Change this to your desired region or set to None for the default region
create_bucket(bucket_name, region)

File: 01-learn-a-language/go/go-basics/task16/create_s3_bucket.go
package main

import (
	"fmt"
	"log"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
)

func createBucket(sess *session.Session, bucketName string) {
	// Create S3 service client
	svc := s3.New(sess)

	// Create the S3 Bucket
	_, err := svc.CreateBucket(&s3.CreateBucketInput{
		Bucket: aws.String(bucketName),
	})
	if err != nil {
		log.Fatalf("Unable to create bucket %q, %v", bucketName, err)
	}

	// Wait until bucket is created before finishing
	fmt.Printf("Waiting for bucket %q to be created...\n", bucketName)
	err = svc.WaitUntilBucketExists(&s3.HeadBucketInput{
		Bucket: aws.String(bucketName),
	})
	if err != nil {
		log.Fatalf("Error occurred while waiting for bucket to be created, %v", bucketName)
	}

	fmt.Printf("Bucket %q successfully created\n", bucketName)
}

func main() {
	// Initialize a session
	sess, err := session.NewSession(&aws.Config{
		Region: aws.String("us-west-1"), // Change this to your preferred region
	})
	if err != nil {
		log.Fatalf("Failed to create session, %v", err)
	}

	// Specify the bucket name
	bucketName := "my-new-bucket-name" // Change this to your desired bucket name

	// Call the function to create the bucket
	createBucket(sess, bucketName)
}
